src/lib/general.c

```diff
@@ -32,18 +32,21 @@
          }
 
 CK_RV general_get_info(CK_INFO *info) {
+    check_pointer(info);
 
     static CK_INFO _info = {
         .cryptokiVersion = CRYPTOKI_VERSION,
-        .manufacturerID = " "TPM2_TOKEN_MANUFACTURER,
+        .manufacturerID = TPM2_TOKEN_MANUFACTURER,
         .flags = 0,
-        .libraryDescription = " "LIBRARY_DESCRIPTION,
+        .libraryDescription = LIBRARY_DESCRIPTION,
         .libraryVersion = {
             /* TODO get from build VERSION */
             .major = 42,
             .minor = 42
         },
     };
+    str_pad(_info.manufacturerID, sizeof(_info.manufacturerID));
+    str_pad(_info.libraryDescription, sizeof(_info.libraryDescription));
 
```

Comment:
If the field is static width can't we do this statically?

Reply:
The problem is as always the 'fill with blanks and must not be null terminated' requirement.
So by assigning it directly, you either have it one  blank to less (and a null terminator), or you are basically storing the null terminator outside of the field.

An alternative is of course to specify each character individually, but this is hard on readability.

Using the method in the patch gets around the problem, as we can simply assign it (=readable) and then we pad it to the correct length and remove the null terminator.